src/onnxruntime.cc

```diff
@@ -1419,9 +1428,30 @@ ModelInstanceState::ValidateInputs(const size_t expected_input_cnt)
                 .c_str());
       }
     } else {
-      RETURN_IF_ERROR(CompareDimsSupported(
-          model_state_->Name(), io_name, iit->second.dims_, dims,
-          model_state_->MaxBatchSize(), false /* compare_exact */));
+      if (model_state_->MaxBatchSize() != 0 || iit->second.dims_.size() > 0) {
+        RETURN_IF_ERROR(CompareDimsSupported(
+            model_state_->Name(), io_name, iit->second.dims_, dims,
+            model_state_->MaxBatchSize(), false /* compare_exact */));
```

Comment:
Can't tell from a glance if this is supported or not, but would it be possible to "scatter" a scalar input to other multi-dimension inputs? 

From client/request side of things, I think Triton core expects that all inputs would have a batch dimension - but I wonder if we can generalize scalar inputs from client side to acceptable "tensor" shapes based on other inputs?

Reply:
discussed offline, an input that only has a batch dimension is already supported.